addons/utm/models/utm_mixin.py

```diff
@@ -59,3 +55,32 @@ def tracking_fields(self):
             ('utm_source', 'source_id', 'odoo_utm_source'),
             ('utm_medium', 'medium_id', 'odoo_utm_medium'),
         ]
+
+    def _find_or_create_record(self, model_name, name):
+        """Based on the URL parameter, retrieve the corresponding record or create it."""
+        Model = self.env[model_name]
+
+        record = Model.search([('name', 'ilike', name)], limit=1)
+
+        if not record:
+            # No record found, create a new one
+            record_values = {'name': name}
+            if 'is_auto_campaign' in record._fields:
+                record_values['is_auto_campaign'] = True
+            record = Model.create(record_values)
+
+        return record
+
+    @api.model
+    def _set_name_unique(self, model_name, name, force_uuid=False):
+        """Try to use the given name for the given model.
+
+        If the name already exists, add random characters at the end.
+        """
+        if not force_uuid:
+            force_uuid = self.env[model_name].search_count([('name', '=', name)])
```

Comment:
Issue is that this basically breaks the batch mode of creation ... for each record you make a search. We should try to keep a batch process, aka search all names then rebuilding counters and updated names based on it. Moreover it would also ease the "force uuid" I think, as everything would be managed here (receiving names + existing and deciding the final counter based on those).